src/main/scala/ReleaseExtra.scala

```diff
@@ -123,16 +123,15 @@ object ReleaseStateTransformations {
     IO.writeLines(file, Seq(versionString))
   }
 
-  private[sbtrelease] lazy val initialVcsChecks = { st: State =>
-    val status = (vcs(st).status !!).trim
-    if (status.nonEmpty) {
-      sys.error("Aborting release. Working directory is dirty.")
+  lazy val initialVcsChecks = (st: State) =>
+    if ((vcs(st).status !!).trim.nonEmpty)
+      sys.error("Aborting release. Working directory has uncommitted changes, " +
+        "please commit your changes before running this command.")
```

Comment:
user can also stash or revert. committing is just one way to solve it
also i remember there were talks about allowing release to continue even wit dirty wd when some setting is set to true